### PR TITLE
Add tests for empty DataFrames

### PR DESCRIPTION
**Pull request by cailinw**
* Test reading in an empty file (blank file with no contents) and writing it out
* (1) but with an empty directory
* Test filtering a CSV dataset (so that its empty) and then writing the result to a file
* Test fFiltering a CSV dataset (so that its empty) and then doing a groupby-subset and writing the result to a file
* (3) and (4) but filtering to a single row
* Test getting a column (which would be a Banyan array) and collecting a reduce (using init) result?

All of the above should for the 3 different kinds of scheduling_configs and for the 3 kinds of files (csv, parquet, arrow)

### CODE REVIEW

File: BanyanDataFrames/test/runtests.jl [hunk 1]
```diff
@@ -216,6 +216,12 @@ function use_stress_data()
     setup_stress_tests()
 end
 
+function use_empty_data()
+    cleanup_tests()
+    setup_basic_tests()
+    setup_empty_tests()
+end
```
Comment (calebwin) [inline BanyanDataFrames/test/runtests.jl:223]: You need to make sure that this function (and other `use_*` functions) can be called multiple times without work being repeatedly done. It doesn't look like `setup_empty_tests` is like that but is it?

**Review by calebwin — CHANGES_REQUESTED** [review]
Looks good overall. Have you tested this on empty data? Some libraries don't really support certain operations - like I don't think HDF5.jl can read in an empty dataset but CSV.jl can read in an empty file... have you checked that what you're testing should actually work?

File: BanyanDataFrames/test/runtests.jl (hunk 1 above)
Comment (cailinw) [reply BanyanDataFrames/test/runtests.jl:223]: Oh, hmm...I don't think `setup_empty_tests` does that. I can change that.

**Comment by cailinw** [comment]
I have tested it in the repl with an empty df, but I haven't tested in reading an empty file (I may have in the past but not recently). Also, these tests don't use HDF5.jl directly, I think. I double check that reading it works

**Review by calebwin — APPROVED** [review]
